main.cpp

```diff
@@ -467,181 +264,86 @@ IntType& IntType::divide(int rhs)
-{
-    value *= (rhs.value);
-    return *this;
-}
-IntType& IntType::divide(const IntType& rhs)
+void part3()
 {
-    if((rhs.value) == 0)
-        std::cout << "floating-point-division-by-zero" <<std::endl;
-    value /= (rhs.value);
-    return *this;
+    FloatType ft( 5.5f );
+    DoubleType dt( 11.1 );
+    IntType it ( 34 );
+    DoubleType pi( 3.14 );
+    
+    std::cout << "The result of FloatType^4 divided by IntType is: " << ft.multiply( ft ).multiply( ft ).divide( it ) << std::endl;
+    std::cout << "The result of DoubleType times 3 plus IntType is : " << dt.multiply( 3 ).add( it ) << std::endl;
+    std::cout << "The result of IntType divided by 3.14 multiplied by DoubleType minus FloatType is: " << it.divide( static_cast<int>(pi) ).multiply( static_cast<int>(dt) ).subtract(static_cast<int>(ft)) << std::endl;
+    std::cout << "An operation followed by attempts to divide by 0, which are ignored and warns user: " << std::endl;
+    std::cout << it.multiply(it).divide(0).divide(0.0f).divide(0.0)<<std::endl;
+    
+    std::cout << "FloatType x IntType  =  " << it.multiply( static_cast<int>(ft) ) << std::endl;
+    std::cout << "(IntType + DoubleType + FloatType) x 24 = " << it.add( static_cast<int>(dt) ).add( static_cast<int>(ft) ).multiply( 24 ) << std::endl;
 }
 
 int main()
-{   
+{
+    
     //testing instruction 0
-    HeapA heapA ; 
+    HeapA heapA ;
 
     //assign heap primitives
     FloatType ft ( 2.0f );
     DoubleType dt ( 2 );
     IntType it ( 2 ) ;
 
-    std::cout << "FloatType add result=" << ( ft.add( 2.0f ).value ) << std::endl;
-    std::cout << "FloatType subtract result=" << ( ft.subtract( 2.0f ).value ) << std::endl;
-    std::cout << "FloatType multiply result=" << ( ft.multiply( 2.0f ).value ) << std::endl;
-    std::cout << "FloatType divide result=" << ( ft.divide( 16.0f).value ) << std::endl << std::endl;
+    std::cout << "FloatType add result=" << ( ft.add( 2.0f ) ) << std::endl;
+    std::cout << "FloatType subtract result=" << ( ft.subtract( 2.0f ) ) << std::endl;
+    std::cout << "FloatType multiply result=" << ( ft.multiply( 2.0f ) ) << std::endl;
+    std::cout << "FloatType divide result=" << ( ft.divide( 16.0f) ) << std::endl << std::endl;
 
-    std::cout << "DoubleType add result=" << ( dt.add(2.0).value ) << std::endl;
-    std::cout << "DoubleType subtract result=" << (dt.subtract(2.0).value) << std::endl;
-    std::cout << "DoubleType multiply result=" << (dt.multiply(2.0).value) << std::endl;
-    std::cout << "DoubleType divide result=" << (dt.divide(5.f).value) << std::endl << std::endl;
+    std::cout << "DoubleType add result=" << ( dt.add(2.0) ) << std::endl;
+    std::cout << "DoubleType subtract result=" << (dt.subtract(2.0) ) << std::endl;
+    std::cout << "DoubleType multiply result=" << (dt.multiply(2.0) ) << std::endl;
+    std::cout << "DoubleType divide result=" << (dt.divide(static_cast<double>(5.f) ) )<< std::endl << std::endl;
```

Comment:
Just change the `float` to a `double` lol!  no need to static_cast that primitive!
- [ ] revise

Reply:
Yeah I did think that was weird but it is hard to know what you allow to be changed and what has to stay the same..